svir/dialogs/standalone_app_dialog.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# /***************************************************************************
+# Irmt
+#                                 A QGIS plugin
+# OpenQuake Integrated Risk Modelling Toolkit
+#                              -------------------
+#        begin                : 2016-06-29
+#        copyright            : (C) 2016 by GEM Foundation
+#        email                : devops@openquake.org
+# ***************************************************************************/
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+from qgis.PyQt.QtCore import (QUrl,
+                              QObject,
+                              pyqtSlot,
+                              QSettings,
+                              )
+from qgis.PyQt.QtGui import (QDialog,
+                             QVBoxLayout,
+                             )
+from qgis.gui import QgsMessageBar
+from svir.ui.gem_qwebview import GemQWebView
+from svir.utilities.shared import DEFAULT_SETTINGS
+
+
+class StandaloneAppDialog(QDialog):
+    """
+    Dialog to be inherited by OpenQuake standalone applications
+
+    :param app_name: short name of the app as it appears in the url
+    :param app_descr: longer name to be used as the window title
+    :param gem_header_name: header that identifies that the application is
+        being driven from QGIS
+    :param gem_header_value: version of the interface between the plugin and
+        the embedded application
+    :param parent: the parent object (optional)
+    """
+
+    def __init__(self, app_name, app_descr, gem_header_name, gem_header_value,
+                 parent=None):
+        super(StandaloneAppDialog, self).__init__(parent)
+
+        self.message_bar = QgsMessageBar(self)
+        self.app_name = app_name
+        self.app_descr = app_descr
+        self.gem_header_name = gem_header_name
+        self.gem_header_value = gem_header_value
+        self.web_view = None
+        self.set_host()
```

Comment:
Why `set_host()` and after each creation (search again set_host around) ?

Reply:
I've added a comment in commit 1f7f740, to exaplain that the user might switch to a different engine server while having embedded applications already initialized and pointing to the previous server, therefore we must let all apps point to the new engine server.